src/components/Card.js

```diff
@@ -1,44 +1,76 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
+    this.isLiked = data.isLiked; //this one
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
-
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
   }
 
   _getTemplate() {
     const cardEl = document
-      .querySelector(this._cardSelector).content
-      .querySelector('.card').cloneNode(true);
+      .querySelector(this._cardSelector)
+      .content.querySelector(".card")
+      .cloneNode(true);
     return cardEl;
   }
 
   _handleLikeButton() {
     // this._cardElement.querySelector(".card__like_button")
     //   .classList.toggle("card__like_button_active");
-      this._likeButton.classList.toggle("card__like_button_active");
+    this._likeButton.classList.toggle("card__like_button_active");
   }
 
   _handleDeleteCard() {
     this._cardElement.remove();
   }
 
+  renderLikeIcon() {
+    if (this.isLiked) {
+      this._likeButton.classList.add("card__like_button_active");
+    } else {
+      this._likeButton.classList.remove("card__like_button_active");
+    }
+  }
+
+  setIsLiked(isLiked) {
+    this.isLiked = isLiked; // this allows to change the value of isLiked
+    this.renderLikeIcon(); // this will update the icon visually
+  }
+
   _setEventListeners() {
     // this._cardElement.querySelector(".card__like_button").addEventListener("click", () => {
     //   this._handleLikeButton();
     // });
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeButton();
+      this._handleLikeButton(); // ?
```

Comment:
`      this._handleLikeButton(); // ?
`

Please, delete it because you do it with `renderLikeIcon` in `index.js`

You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.